velox/connectors/hive/storage_adapters/hdfs/HdfsFileSystem.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <hdfs/hdfs.h>
+#include "HdfsReadFile.h"
+#include "velox/common/file/FileSystems.h"
+
+namespace facebook::velox::filesystems {
+struct HdfsServiceEndpoint {
+  std::string host;
+  int port;
+};
+
+class HdfsFileSystem : public FileSystem {
+ private:
+  hdfsFS hdfsClient_;
```

Comment:
Should this be a `std::shared_ptr`?

Reply:
hdfsFS is already a pointer.  
The hdfs library defines it like this: `typedef struct HdfsFileSystemInternalWrapper * hdfsFS;`
I dont think we should wrap it again in a shared pointer.